packages/@aws-cdk/aws-synthetics/test/canary.test.ts

```diff
@@ -0,0 +1,223 @@
+    }),
+  }))
+    .toThrowError('Canary name must be lowercase, numbers, hyphens, or underscores (no spaces) (myCanary)');
+});
+
+test('Throws when name has more than 21 characters', () => {
+  expect(() => new synthetics.Canary(stack, 'Canary', {
+    canaryName: 'a'.repeat(22),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  }))
+    .toThrowError('Canary name is too large, must be <= 21 characters, but is 22');
+});
+
+test('An existing role can be specified instead of auto-created', () => {
+  // GIVEN
+  const role = new iam.Role(stack, 'role', {
+    assumedBy: new iam.ServicePrincipal('lambda.amazonaws.com'),
+  });
+
+  // role.addToPolicy(/* required permissions per the documentation */);
+
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    role,
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    ExecutionRoleArn: stack.resolve(role.roleArn),
+  });
+});
+
+test('An existing bucket can be specified instead of auto-created', () => {
+  // GIVEN
+  const bucket = new s3.Bucket(stack, 'mytestbucket');
+  const prefix = 'canary';
+
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    artifactsBucketLocation: { bucket, prefix },
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    ArtifactS3Location: stack.resolve(bucket.s3UrlForObject(prefix)),
+  });
+});
+
+test('Runtime can be specified', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    runtime: synthetics.Runtime.SYNTHETICS_1_0,
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Runtime can be customized', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    runtime: new synthetics.Runtime('syn-1.0'),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    RuntimeVersion: 'syn-1.0',
+  });
+});
+
+test('Schedule can be set with Rate', () => {
+  // WHEN
+  new synthetics.Canary(stack, 'Canary', {
+    schedule: synthetics.Schedule.rate(Duration.minutes(3)),
+    test: synthetics.Test.custom({
+      handler: 'index.handler',
+      code: synthetics.Code.fromInline('foo'),
+    }),
+  });
+
+  // THEN
+  expect(stack).toHaveResourceLike('AWS::Synthetics::Canary', {
+    Schedule: objectLike({ Expression: 'rate(3 minutes)'}),
+  });
```

Comment:
You probably should add a test to validate the correct expression is generated for `rate(1 minute)`.